addons/l10n_in/models/account_invoice.py

```diff
@@ -45,13 +45,16 @@ def _compute_l10n_in_gst_treatment(self):
                 record.l10n_in_gst_treatment = gst_treatment
         (self - indian_invoice).l10n_in_gst_treatment = False
 
-    @api.depends('partner_id', 'company_id')
+    @api.depends('partner_id', 'partner_shipping_id', 'company_id')
     def _compute_l10n_in_state_id(self):
         for move in self:
             if move.country_code == 'IN' and move.journal_id.type == 'sale':
                 country_code = move.partner_id.country_id.code
                 if country_code == 'IN':
-                    move.l10n_in_state_id = move.partner_id.state_id
+                    if move.partner_id == move.partner_shipping_id.parent_id and move.partner_shipping_id.type == 'delivery':
+                        move.l10n_in_state_id = move.partner_shipping_id.state_id
```

Comment:
What if `move.partner_shipping_id.state_id` is False ? Shouldn't we have a fallback ? Perhaps a question for the PO

Reply:
@RicGR98 Hey, I have made the changes. Please have a look